Add CSV max filesize check (#190)

daglueck opened the pull request:
:heart: Thank you for your contribution!

### Description

Fixes #190 by means of adding a file size check to `invenio_previewer.extensions.csv_papaparsejs.can_preview`.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

lnielsen commented:
@psaiz Question: The new CSV file previewer is supposed to stream the CSV file so it's not loading the full file right? 

@daglueck Given the answer from @psaiz I would say that the limit should be significantly higher to allow for larger CSV files to be previewed - say e.g. 100MB.

alejandromumo commented on invenio_previewer/extensions/csv_papaparsejs.py:
```diff
@@ -17,7 +17,14 @@
 def can_preview(file):
     """Determine if the given file can be previewed."""
-    return file.is_local() and file.has_extensions(".csv", ".dsv")
+    max_file_size = current_app.config.get(
```
Would it make sense to have a validator function (i.e. `validate_csv`) to separate concerns? That way, it would be easier to extend in the future in case more rules are needed.

This is done for other extensions (e.g. [json](https://github.com/inveniosoftware/invenio-previewer/blob/141038234f0d7a64a28cc651b3b65c8ff58eef74/invenio_previewer/extensions/json_prismjs.py))

daglueck commented:
Based on your comments, I've added a separate validation function (@alejandromumo ) and increased the max file size to `100 * 1024 * 1024` (@lnielsen ). I've added a new config variable `PREVIEWER_CSV_MAX_BYTES` to differentiate CSV from JSON/XML. Let me know if you had imagined this in a different way.

@lnielsen I didn't know that a new csv previewer was added. How does it work with the max file size, will it simply not load the file if it's too big (as with XML/JSON I guess) or will it truncate the file (like with TXT)? I was assuming here it won't load the file since otherwise there might be parsing issues. If on the other hand it does truncate the file, the wording of `PREVIEWER_CSV_MAX_BYTES` might need to be adjusted.

ntarocco commented:
Thanks a lot for the contribution!
